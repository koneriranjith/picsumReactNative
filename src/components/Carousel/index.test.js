import 'react-native';
import React from 'react';
import Index from './index';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

let mockData = {
    horizontal: true,
    pagingEnabled: true,
    scrollEnabled: true,
    snapToAlignment: 'center',
    onEndReachedThreshold: 0.5,
    scrollEventThrottle: 16,
    decelerationRate: 'fast',
    showsHorizontalScrollIndicator: true,
    showsHorizontalScrollIndicatorLimit: 10,
    data: [{src:'1'},{src:'1'}],
    onEndReached: jest.fn() 
}

test('should match snapShot', ()=>{
    const container = renderer.create(<Index />).toJSON();
    expect(container).toMatchSnapshot();
    
})

test('should render Text loading without Data', ()=>{
    const container = shallow(<Index />).props();
    expect(container.children.props.children).toEqual('loading...');
})

test('should render component with Data', ()=>{
    const container = renderer.create(<Index {...mockData}/>).toJSON();
    expect(container).toMatchSnapshot()
    expect(container.children[1].children.length).toEqual(10);
})

test('should render Image', ()=>{
    const container = shallow(<Index {...mockData}/>).props();
    console.log('In final res',container)
    expect(container.find('img').length).toEqual(1);
})
