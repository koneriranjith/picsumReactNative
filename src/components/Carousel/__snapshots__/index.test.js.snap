// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapShot 1`] = `
<View>
  <Text>
    loading...
  </Text>
</View>
`;

exports[`should render component with Data 1`] = `
<View>
  <RCTScrollView
    data={
      Array [
        Object {
          "src": "1",
        },
        Object {
          "src": "1",
        },
      ]
    }
    decelerationRate="fast"
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.5}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onViewableItemsChanged={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={16}
    showsHorizontalScrollIndicator={true}
    snapToAlignment="center"
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfig={
      Object {
        "viewAreaCoveragePercentThreshold": 50,
      }
    }
    viewabilityConfigCallbackPairs={
      Array [
        Object {
          "onViewableItemsChanged": [Function],
          "viewabilityConfig": Object {
            "viewAreaCoveragePercentThreshold": 50,
          },
        },
      ]
    }
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "elevation": 5,
              "flex": 1,
              "height": 444.6666666666667,
              "margin": 10,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 0.5,
                "width": 0.5,
              },
              "shadowOpacity": 0.5,
              "shadowRadius": 3,
              "width": 730,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "fontSize": 22,
                }
              }
            />
          </View>
          <Image
            source={
              Object {
                "uri": "https://picsum.photos/200/300?image=undefined",
              }
            }
            style={
              Object {
                "height": 444.6666666666667,
                "width": 730,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "elevation": 5,
              "flex": 1,
              "height": 444.6666666666667,
              "margin": 10,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 0.5,
                "width": 0.5,
              },
              "shadowOpacity": 0.5,
              "shadowRadius": 3,
              "width": 730,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "fontSize": 22,
                }
              }
            />
          </View>
          <Image
            source={
              Object {
                "uri": "https://picsum.photos/200/300?image=undefined",
              }
            }
            style={
              Object {
                "height": 444.6666666666667,
                "width": 730,
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#595959",
          "borderRadius": 5,
          "height": 10,
          "margin": 8,
          "opacity": 1,
          "width": 10,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#595959",
          "borderRadius": 5,
          "height": 10,
          "margin": 8,
          "opacity": 0.3,
          "width": 10,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#595959",
          "borderRadius": 5,
          "height": 10,
          "margin": 8,
          "opacity": 0.3,
          "width": 10,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#595959",
          "borderRadius": 5,
          "height": 10,
          "margin": 8,
          "opacity": 0.3,
          "width": 10,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#595959",
          "borderRadius": 5,
          "height": 10,
          "margin": 8,
          "opacity": 0.3,
          "width": 10,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#595959",
          "borderRadius": 5,
          "height": 10,
          "margin": 8,
          "opacity": 0.3,
          "width": 10,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#595959",
          "borderRadius": 5,
          "height": 10,
          "margin": 8,
          "opacity": 0.3,
          "width": 10,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#595959",
          "borderRadius": 5,
          "height": 10,
          "margin": 8,
          "opacity": 0.3,
          "width": 10,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#595959",
          "borderRadius": 5,
          "height": 10,
          "margin": 8,
          "opacity": 0.3,
          "width": 10,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#595959",
          "borderRadius": 5,
          "height": 10,
          "margin": 8,
          "opacity": 0.3,
          "width": 10,
        }
      }
    />
  </View>
</View>
`;
